File: testdata/jennies/rawtypes/enums/PHPRawTypes/src/Enums/DashboardCursorSync.php
<?php

namespace Grafana\Foundation\Enums;

/**
 * 0 for no shared crosshair or tooltip (default).
 * 1 for shared crosshair.
 * 2 for shared crosshair AND shared tooltip.
 */
final class DashboardCursorSync implements \JsonSerializable, \Stringable {
    /**
     * @var int
     */
    private $value;

    /**
     * @var array<string, DashboardCursorSync>
     */
    private static $instances = [];

    private function __construct(int $value)
    {
        $this->value = $value;
    }

    public static function off(): self
    {
        if (!isset(self::$instances["Off"])) {
            self::$instances["Off"] = new self(0);
        }

        return self::$instances["Off"];
    }

    public static function crosshair(): self
    {
        if (!isset(self::$instances["Crosshair"])) {
            self::$instances["Crosshair"] = new self(1);
        }

        return self::$instances["Crosshair"];
    }

    public static function tooltip(): self
    {
        if (!isset(self::$instances["Tooltip"])) {
            self::$instances["Tooltip"] = new self(2);
        }

        return self::$instances["Tooltip"];
    }

    public static function fromValue(int $value): self
    {
        if ($value === 0) {
            return self::off();
        }

        if ($value === 1) {
            return self::crosshair();
        }

        if ($value === 2) {
            return self::tooltip();
        }

        throw new \UnexpectedValueException("Value '$value' is not part of the enum DashboardCursorSync");
    }

    public function jsonSerialize(): int
    {
        return $this->value;
    }

    public function __toString(): string
    {
        return (string) $this->value;
    }
}

File: testdata/jennies/rawtypes/enums/JavaRawTypes/enums/DashboardCursorSync.java
package enums;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonValue;


// 0 for no shared crosshair or tooltip (default).
// 1 for shared crosshair.
// 2 for shared crosshair AND shared tooltip.
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum DashboardCursorSync {
    OFF(0),
    CROSSHAIR(1),
    TOOLTIP(2);

    private final Integer value;

    private DashboardCursorSync(Integer value) {
        this.value = value;
    }

    @JsonValue
    public Integer Value() {
        return value;
    }
}
